### PR TITLE
DOC clean out the documentation

### LINKED ISSUE #617: Scattered docs
Our documentation is currently scattered in
- the CCL note
- multiple READMEs
- code examples (C and python)
- the wiki
- the CCL paper
- doxygen
- readthedocs

We should consolidate docs in as much as possible. The proposal is to put python code example snippets in the README, and ensure we maintain this as well as readthedocs and the README in the benchmark folder.

In this proposal, the CCL note and wiki would be deprecated (with the relevant information moved to the README) and the jupyter notebooks would move to a dedicated repository.

### PR DESCRIPTION
**Pull request by beckermr**
OK friends! I have cleaned out all of the documentation into a single coherent thing.

Closes #617

### CODE REVIEW

File: readthedocs/source/navigating_the_code.rst [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+- ``ccl_f2d.c``: 2-d interpolations in ``C``
+- ``ccl_power.c``: code to spline and initialize the power spectra and
+  transfer functions
+- ``ccl_bcm.c``: code to compute the BCM model for baryonic effects
+- ``ccl_eh.c``: code to compute the Eisenstein and Hu (1998) transfer function
+  approximation
+- ``ccl_musigma.c``: code to properly normalize input linear power spectra for the
+  ``mu-Sigma`` modified gravity model
+- ``ccl_cls.c``: code to compute angular power spectra from 3-d power spectra
+- ``ccl_massfunct.c``: code for the halo mass function and halo bias models, also
```
Comment (elisachisari) [inline readthedocs/source/navigating_the_code.rst]: ccl_massfunct -> ccl_massfunc? Or perhaps this changed recently.

File: readthedocs/source/installation.rst [hunk 2]
```diff
@@ -1,273 +1,109 @@
...
-
-Additionally, to build the code you will need
-
-* `CMake <https://cmake.org/>`_ version 3.2 or above.
-* `SWIG <http://www.swig.org/>`_ is needed if you wish to modify CCL and have it availabe in Python.
-
-:code:`CMake` is the only requirement that needs to be installed manually if you are using :code:`pip`.
-
-On Ubuntu:
+Installing CCL with ``conda`` will also install a copy of ``pycamb``, the
```
[the remark this answers is not in the document]
Comment (beckermr) [reply readthedocs/source/installation.rst]: yes!

File: readthedocs/source/installation.rst [hunk 3]
```diff
@@ -1,273 +1,109 @@
...
-========
-
-CCL is available as a Python package through PyPi. To install, simply run:
-
-.. code-block:: bash
-
-   $ pip install pyccl
+CCL can be installed from ``pip``, ``conda``, or directly from source.
+It is configured to install most of its requirements automatically. However, if
+you want to use CCL with Boltzmann codes like ``CLASS`` or ``CAMB``, you will
```
[the remark this answers is not in the document]
Comment (beckermr) [reply readthedocs/source/installation.rst]: yes - with conda this comes in automatically, but in general they need to be installed by hand.

File: readthedocs/source/installation.rst [hunk 4]
```diff
@@ -1,273 +1,109 @@
...
+----
 
-   $ python setup.py install # append --user for single user install
+If you are working in a ``conda`` environment, then ``CAMB`` is available via
 
-The pyccl module will be installed into a sensible location in your :code:`PYTHONPATH`, and so should be picked up automatically by your Python interpreter. You can then simply import the module using import pyccl.
+.. code-block:: bash
 
-You can quickly check whether pyccl has been installed correctly by running :code:`python -c "import pyccl"` and checking that no errors are returned.
+   $ conda install -c conda-forge camb
```
[the remark this answers is not in the document]
Comment (beckermr) [reply readthedocs/source/installation.rst]: yup. I added a note to this effect.

File: readthedocs/source/citation.rst [hunk 5]
```diff
@@ -1,6 +1,19 @@
...
+the ideas or software here, please cite that paper and provide a link to the repository
+``https://github.com/LSSTDESC/CCL``. You are welcome to re-use the code, which is open source and
+available under terms consistent with our `LICENSE <https://github.com/LSSTDESC/CCL/blob/master/LICENSE>`_
+(`BSD 3-Clause <https://opensource.org/licenses/BSD-3-Clause>`_).
 
+External contributors and DESC members wishing to use CCL for non-DESC projects
+should consult with the TJP working group conveners, ideally before the work has
+started, but definitely before any publication or posting of the work to the arXiv.
+
+For free use of the ``CLASS`` library, the ``CLASS`` developers require that the
```
[the remark this answers is not in the document]
Comment (beckermr) [reply readthedocs/source/citation.rst]: None that I can find, though OFC it is nice. I will add a note to their license which has some clauses in it.

**Comment by beckermr** [comment]
Thanks @elisachisari! The installation problems on the wiki are outdated. My plan is to delete a bunch of stuff on the wiki once v2 goes live or at least mark it as applying to v1 only.

**Comment by beckermr** [comment]
Otherwise, all of the changes have been addressed!

**Comment by elisachisari** [comment]
Regarding the outdated info in the wiki, I think we should keep a section on v1 for a few months, but I agree with making v2 the default.

**Pull request merged**